Non-Integer inputs

sammy00747 opened the pull request:
Changes have been made in inc/helper.h and src/main.c
In this pull request, non-integer inputs have been taken care..
Here, position input is taken as string..
And if there are any non-digit character in the string (also if the string 
has more than 2 or no character) then the input is not accepted.
Else the input is accepted and converted to corresponding integer..
Hope it helps..

sammy00747 committed:
shortlog: Correction for non-integer inputs

Though this may seem as a small bugfix, I have
added a new file - test.c - which is independent of all the other files
in the repositiory

fixes #43

Signed-off by: [Sumit Mridha] 3sumit5@gmail.com

sammy00747 committed:
shortlog: Example pointed out

the piece of code which is a solution is there from line 26 to line 57
this will help in understanding the solution

fixes #43

Signed-off by: [Sumit Mridha] 3sumit5@gmail.com

sammy00747 committed:
Changed src/main.c

Removed the assert() function, as there were possibilities that non-integer
inputs can be given for the position. Instead input is taken as string.

Signed-off-by: Sumit Kumar Mridha <3sumit5@gmail.com>

RudraNilBasu commented on src/main.c:
```diff
@@ -25,18 +25,57 @@ int main(int argc, char* argv[])
 	}
 
-	int result = 0, pos, player = 0;
+	int result = 0, pos, player = 0,i,sw=0;
```
Use a more meaningful name instead of `sw` (same for most other newly created variables)

sammy00747 replied:
naam di66i avoid_error..

RudraNilBasu commented on src/main.c:
```diff
@@ -25,18 +25,57 @@ int main(int argc, char* argv[])
-			printf("Enter Position: "); 
-			assert(scanf("%d", &pos) == 1);
+			printf("Enter Position (non-digit characters not allowed.. not even white spaces...): ");
+			goto init__1;
```
Using `goto` is a bad practice. Still to be confirmed with @Progyan1997

0xTheProDev replied:
We're strictly against of use of `goto`. Try alternatives if possible. @sammy00747

sammy00747 replied:
not possible.. atleast, i don't know..

RudraNilBasu commented on src/main.c:
```diff
@@ -25,18 +25,57 @@ int main(int argc, char* argv[])
+				c = getchar();
+				s[i++] = c;
+				s = realloc(s,i+1);
+			}while(c != '\n' && c != EOF);
```
Follow the coding style: https://bytesclub.github.io/docs/coding_style.pdf

RudraNilBasu commented on src/main.c:
```diff
@@ -25,18 +25,57 @@ int main(int argc, char* argv[])
+					printf(ANSI_YELLOW "Wrong input.. Enter again...\n" ANSI_RESET);
+					goto error_pos;
+				}
+				else
```
Same here.

RudraNilBasu reviewed (changes requested):
See inline comments

sammy00747 committed:
Minor changes

Changes have been made, keeping in mind the coding style of Bytes Club

Signed-off-by: Sumit Kumar Mridha <3sumit5@gmail.com>

sammy00747 committed:
Variable name change

The variable name 'sw' has been changed to
'avoid-error' for meaningfulness.

Signed-off-by: Sumit Kumar Mridha <3sumit5@gmail.com>

sammy00747 committed:
Codinbg style

Changes have been made, according to the recommended
coding style.
Header file ctype.h removed, and Isdigit() is
defined in inc/header.h and src/header.c .

Signed-off-by: Sumit Kumar Mridha <3sumit5@gmail.com>
